pkg/executor/aggregate/agg_hash_final_worker.go

```diff
@@ -46,11 +47,32 @@ type HashAggFinalWorker struct {
 	partialResultMap    aggfuncs.AggPartialResultMapper
 	BInMap              int
 	isFirstInput        bool
-	groupSet            set.StringSetWithMemoryUsage
 	inputCh             chan *aggfuncs.AggPartialResultMapper
 	outputCh            chan *AfFinalResult
 	finalResultHolderCh chan *chunk.Chunk
 	groupKeys           [][]byte
+
+	spillHelper        *parallelHashAggSpillHelper
+	isSpilledTriggered bool
+	spilledDataChan    chan *aggfuncs.AggPartialResultMapper
+
+	// These agg functions are partial agg functions that are same with partial workers'.
+	// They only be used for restoring data that are spilled to disk in partial stage.
+	aggFuncsForRestoring []aggfuncs.AggFunc
```

Comment:
1. put this definition into `parallelHashAggSpillHelper`
2. add a memTracker in spillHelper and attach this memTracker to HashAggExec
3. define `restoreFromOneSpillFile`, `restoreOnePartition`, `processRow` as the member attributes of spillHelper

Reply:
> 1. put this definition into `parallelHashAggSpillHelper`
> 2. add a memTracker in spillHelper and attach this memTracker to HashAggExec
> 3. define `restoreFromOneSpillFile`, `restoreOnePartition`, `processRow` as the member attributes of spillHelper

done